### PR TITLE
14/add organization authentication

### LINKED ISSUE #14: Add authentication for organizations
A user should be able to log in as their organization to access the application.

To start with, every user will be logged in under the same account, which represents the organization as a whole. Every user will be able to do all operations. If we have time, we will develop a user-based authentication system with granular permissions.

Authentication will be with email and password. For now, the organization will share a single password.

See #13 to follow development of a full authentication system with users.

### PR DESCRIPTION
**Pull request by AdamVig**
Closes #14.

This PR will cause test coverage to decline. #54 addresses test coverage for this feature.

### CODE REVIEW

File: controllers/auth.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    return next(new restify.UnprocessableEntityError('missing required fields'))
+  }
+}
+
+// Initialize Passport middleware
+module.exports.initialize = passport.initialize()
+
+// Register an organization
+module.exports.register = (req, res, next) => {
+  if (req.body.name && req.body.email && req.body.password) {
```
Comment (emroussel) [inline controllers/auth.js:55]: Should there be a check after this to make sure the email is not already used and send an appropriate error message if it is the case?

**Review by emroussel — CHANGES_REQUESTED** [review]
Can you add a section in the README describing how to add the authentication header to requests (mostly for the format)?

File: controllers/auth.js (hunk 1 above)
Comment (AdamVig) [reply controllers/auth.js:55]: Good point. I think I might add a unique constraint to the email column in the database, ensuring that an error will be returned if an email is non-unique. What do you think?

Comment (emroussel) [reply controllers/auth.js:55]: Yeah good idea. Would that return the appropriate error code from the api though? (I think it should be a 409)

Comment (AdamVig) [reply controllers/auth.js:55]: Good point! I'll catch the error from the database and return the appropriate error code.

Comment (AdamVig) [reply controllers/auth.js:55]: Just want to point out that this functionality already exists, so I won't actually have to add a commit to this PR in order for it to work (I outsmarted myself): [see db.js](https://github.com/AdamVig/stockpile-api/blob/master/services/db.js#L26). 😃

**Review by emroussel — APPROVED** [review]

**Pull request merged**